src/coprocessor/dag/expr/builtin_string.rs

```diff
@@ -1869,4 +1931,92 @@ mod tests {
             assert_eq!(got, Datum::Bytes(exp.as_bytes().to_vec()));
         }
     }
+
+    #[test]
+    fn test_substring_binary_2_args() {
+        let tests = vec![
+            ("中文a测试bb", 1, "中文a测试bb"),
```

Comment:
I think you can also add some tests which use hex to write some binary literal. It will be more obvious to see that it is in fact doing binary substring.

Reply:
https://github.com/tikv/tikv/blob/56733c60961608392a230e6a6de2048176674799/src/coprocessor/dag/expr/builtin_string.rs#L1958-L1960
hi, is this  multibytes test what you want to see? or I write it as 
`(vec![0x61, 0xE6, 0xB5, 0x8B, 0xE8, 0xAF, 0x95], Datum::I64(-1), vec![149]),`?

Reply:
Yes, or you can write them like "\x61\xE6", since it is not very obvious that a single Chinese character will be mapped to 3 bytes (requires some background knowledge to know how it works)

Reply:
some test cases added,  PTAL, thanks